File: src/main/java/net/pterodactylus/sone/freenet/wot/IdentityManager.java
package net.pterodactylus.sone.freenet.wot;

import java.util.Set;

import net.pterodactylus.util.service.Service;

import com.google.common.eventbus.EventBus;
import com.google.inject.ImplementedBy;

/**
 * Connects to a {@link WebOfTrustConnector} and sends identity events to an
 * {@link EventBus}.
 *
 * @author <a href="mailto:bombe@pterodactylus.net">David ‘Bombe’ Roden</a>
 */
@ImplementedBy(IdentityManagerImpl.class)
public interface IdentityManager extends Service {

	boolean isConnected();
	Set<OwnIdentity> getAllOwnIdentities();

}

File: src/main/kotlin/net/pterodactylus/sone/freenet/wot/IdentityManager.kt
package net.pterodactylus.sone.freenet.wot

import net.pterodactylus.util.service.Service

import com.google.common.eventbus.EventBus
import com.google.inject.ImplementedBy

/**
 * Connects to a [WebOfTrustConnector] and sends identity events to an
 * [EventBus].
 */
@ImplementedBy(IdentityManagerImpl::class)
interface IdentityManager : Service {

	val isConnected: Boolean
	val allOwnIdentities: Set<OwnIdentity>

}
